sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -212,7 +210,15 @@ def _resolve_keyvault_reference(
 
     if referenced_client is None and key_vault_options.credential is not None:
         referenced_client = SecretClient(
-            vault_url=key_vault_identifier.vault_url, credential=key_vault_options.credential
+            vault_url=key_vault_identifier.vault_url, credential=key_vault_options.credential, 
+             **key_vault_options.client_configs.get(key_vault_identifier.vault_url, {})
+        )
+        provider._secret_clients[key_vault_identifier.vault_url] = referenced_client
+    elif referenced_client is None:
+        options = key_vault_options.client_configs[key_vault_identifier.vault_url]
+        referenced_client = SecretClient(
+            vault_url=key_vault_identifier.vault_url, credential=options.pop("credential"), 
+             **options
```

Comment:
Line 209 is no longer applicable, as `provider._secret_clients` is not longer being filled earlier, so `referenced_client` will always be None. We should also prioritize the specific credential over the 'global' credential.
On reflection, I think we can simplify the code after line 206 with something like:
```python
keyvault_credential = key_vault_options.client_configs.get(key_vault_identifier.vault_url, key_vault_options.credential)
if keyvault_credential is not None:
    secret_client = SecretClient(
        vault_url=key_vault_identifier.vault_url,
        credential=keyvault_credential,
        **key_vault_options.client_configs.get(key_vault_identifier.vault_url, {})
    return secret_client.get_secret(key_vault_identifier.name, version=key_vault_identifier.version).value
if key_vault_options.secret_resolver is not None:
    return key_vault_options.secret_resolver(config.secret_id)
raise ValueError(
    "No Secret Client found for Key Vault reference %s" % (key_vault_identifier.vault_url)
)
```

Reply:
209 is still needed. It is only None the first time. We reuse the same client for every connection to the same key vault. Each key vault reference only points to a single key, and Key Vault only allows accessing one secret at a time.